pkg/user/rpc_user.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package user
+
+import (
+	"context"
+	"encoding/gob"
+	"net/rpc"
+
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/plugin"
+	hcplugin "github.com/hashicorp/go-plugin"
+)
+
+func init() {
+	gob.Register(&userpb.User{})
```

Comment:
We can register interfaces (which we would like to pass as context) here. But I am not sure what all structs/interfaces that we'd like to add!

Reply:
Are we able to log from the plugins now?

Reply:
I tried passing the user object as a part of the context. I was able to access the user object (via the context.Value) on the plugin side. But here's the problem with the logger: the appctx's `WithLogger` attaches zerolog's `ctxKey` struct to the context, which is not exported, hence we cannot register that at init time.

Reply:
Can you look into ways of getting around that?

Reply:
In the auth pkg, we're passing `auth.Scope` without registering it. How does that work?

Reply:
I think this *might* be a roadblock as there is not way to `gob.Register(ctxkey{})` at init stage because ctxKey{} is private! Hence the RPC is always going to fail when attaching logger to context.

Reply:
> Can you look into ways of getting around that?

Yes. I was just investigating ways around this!

Reply:
I'll merge this for now but this needs to be addressed ASAP.

Reply:
Yes! So the reason we're getting the gob error is because we're passing `map[interface{}]interface{}`, If we are passing `interface{}` through gob, that NEEDS to be registered, passing structs is fine. That's why it does not complain when using `auth.Scope`. https://pkg.go.dev/encoding/gob#Register I'll further dig into this

Reply:
Then instead of the map, let's store all the keys that we know of in a struct. The ones we currently use are the logger, user object and the token.